packages/gatsby/src/redux/persist.ts

```diff
@@ -81,10 +81,20 @@ function guessSafeChunkSize(values: [string, IGatsbyNode][]): number {
   const valueCount = values.length
   const step = Math.max(1, Math.ceil(valueCount / nodesToTest))
   let maxSize = 0
+  let warnFlag = false;
+
   for (let i = 0; i < valueCount; i += step) {
     const size = v8.serialize(values[i]).length
+    if (size > 500000) {
```

Comment:
The `maxSize` already tracks the size of the largest node, so why would you check this inside the loop, and not check `maxSize` instead after the loop?